File: Java/GradingStudents.java
import java.util.*;

public class GradingStudents {
    public static List<Integer> gradingStudents(List<Integer> grades) {
        int length = grades.size();
        List<Integer> finalGrade = new ArrayList<Integer>();
        
        for (int i = 0; i < length; i++) {
            int diff = (int) Math.ceil(grades.get(i) / 5.0) * 5 - grades.get(i);
            if (diff < 3 && grades.get(i) >= 38) {
                finalGrade.add((int) Math.ceil(grades.get(i) / 5.0) * 5);
            } else {
                finalGrade.add(grades.get(i));
            }
        }
        return finalGrade;
    }
    
    public static void main(String[] args) {
        List<Integer> grades = Arrays.asList(73, 67, 38, 33);
        List<Integer> finalGrade = gradingStudents(grades);
        System.out.println(finalGrade);
    }
}

File: JavaScript/gradingStudents.js
function gradingStudents(grades) {
    let length = grades.length;
    let finalGrade = [];

    for (let i = 0; i < length; i++) {
        let diff = Math.ceil(grades[i] / 5) * 5 - grades[i];
        if (diff < 3 && grades[i] >= 38) {
            finalGrade.push(Math.ceil(grades[i] / 5) * 5)
        } else {
            finalGrade.push(grades[i]);
        }
    }
    return finalGrade;


}


console.log(gradingStudents([73, 67, 38, 33]))